invenio_records_resources/resources/files/resource.py

```diff
@@ -180,6 +182,12 @@ def read_content(self):
         if item is None:
             abort(404)
 
+        # emit file download stats event
+        obj = item._file.object_version
+        emitter = build_event_emitter("file-download")
```

Comment:
minor: since this is re-usable, can it be initialized only once e.g. as a cached property, or another "one-time" init/setup place?

Reply:
sure – the built `EventEmitter` for a specific event name can be re-used all over the place
given that these emitters are used both here as well as app-rdm, should we maybe put the caching logic into invenio-stats itself?
e.g. on the extension object, something along the lines of `ext.event_emitters[event_name]`?

Reply:
Oof, forgot to reply, but yeah `invenio-stats` would be the place, and the current implementation looks good :)